scripts/billing/sql/daily_compute_usage_cud_sud_coverage.sql

```diff
@@ -0,0 +1,156 @@
+  (
+    SELECT
+      u.usage_date,
+      u.location.region AS region,
+      u.project.id AS project_id,
+      u.sku.id AS sku_id,
+      u.sku.description AS sku_description,
+      IFNULL(system_labels.value, "UNKNOWN") AS machine_spec,
+      usage.unit,
+      prices.unit_price,
+      cred.name,
+      IF (
+        prices.unit_price = 0 OR lower(cred.name) NOT LIKE "committed%",
+        0, 
+        CASE
+          -- Divide credit $ amount by unit price to calculate amount of usage offset by credit
+          WHEN LOWER(usage.unit) LIKE "seconds" THEN -1*SUM(cred.amount)/prices.unit_price      
+          WHEN LOWER(usage.unit) = "byte-seconds" THEN -1*SUM(cred.amount)/prices.unit_price
+          ELSE NULL
+        END
+      ) AS CUD_covered_usage,
+      IF (lower(cred.name) LIKE "committed%", SUM(cred.amount),0) AS CUD_cost,
+      IF (
+        prices.unit_price = 0 OR LOWER(cred.name) NOT LIKE "sustained%",
+        0, 
+        CASE
+          -- Divide credit $ amount by unit price to calculate amount of usage offset by credit
+          WHEN LOWER(usage.unit) LIKE "seconds" THEN -1*SUM(cred.amount)/prices.unit_price
+          WHEN LOWER(usage.unit) = "byte-seconds" THEN -1*SUM(cred.amount)/prices.unit_price
+          ELSE NULL
+        END
+      ) AS SUD_covered_usage,
+      IF (lower(cred.name) LIKE "sustained%", SUM(cred.amount),0) AS SUD_cost
+    FROM usage_data AS u, UNNEST(credits) AS cred
+    LEFT JOIN UNNEST(system_labels) AS system_labels
+      ON system_labels.key = "compute.googleapis.com/machine_spec"
+    INNER JOIN prices ON TRUE
+      AND u.sku.id = prices.sku_id
+      AND u.location.region = prices.region
+      AND u.usage_date = prices.usage_date
+    GROUP BY 1,2,3,4,5,6,7,8,9
+    ORDER BY 1,2,3,4,5,6,7,8,9 DESC
+  )
+  GROUP BY 1,2,3,4,5,6
+),
+
+-- Temporary table containing usage amount and usage cost 
+-- before credit based discounts (e.g. CUD, SUD).
+cost_data AS
+(
+  SELECT
+    u.usage_date,
+    u.location.region AS region,
+    u.usage.unit AS unit,
+    u.project.id AS project_id,
+    u.project.name AS project_name,
+    u.sku.id AS sku_id,
+    u.sku.description AS sku_description,
+    IFNULL(system_labels.value, "UNKNOWN") AS machine_spec,
+    SUM(usage.amount) AS usage_amount,
+    SUM(cost) AS cost
+  FROM usage_data AS u
+  LEFT JOIN UNNEST(system_labels) as system_labels
+    ON system_labels.key = "compute.googleapis.com/machine_spec"
+  GROUP BY 1,2,3,4,5,6,7,8
+  ORDER BY 1,2,3,4,5,6,7,8 ASC
+)
+
+-- Final select statement to join everything together into one dataset
+SELECT
+  a.usage_date,
+  a.region,
+  a.unit,
+  a.project_id,
+  a.project_name,
+  a.sku_id,
+  a.sku_description,
+  a.machine_spec,
+  SUM(a.usage_amount) AS usage_amount,
+  SUM(a.cost) AS cost,
+  SUM(CUD_covered_usage) AS CUD_covered_usage,
+  SUM(CUD_cost) AS CUD_cost,
+  SUM(SUD_covered_usage) AS SUD_covered_usage,
+  SUM(SUD_cost) AS SUD_cost
+FROM cost_data a
+FULL OUTER JOIN credit_data b ON TRUE
```

Comment:
Why `ON TRUE`?

Reply:
It just allows me to list out the ANDs below all with ANDs without having to worry about making the first line different (e.g. w/ an ON instead of an AND). That being said, it's clearly not part of the style guide, so I'll go back and change it.